src/main/kotlin/com/hedvig/accountservice/termination/TerminationServiceImpl.kt

```diff
@@ -35,130 +27,70 @@ class TerminationServiceImpl(
-        } catch (exception: Exception) {
-            null
-        }
-
-        val personFlags: PersonalFlagResponse =
-            memberService.getPersonFlagsByMemberId(
-                memberId,
-                market
-            )
-
-        val memberTerminationOutcome = when {
-            numberFailedCharges >= numberOfFailedChargesRequiredForTermination(market) -> {
-                MemberTerminationOutcome.Terminated(
-                    memberId = memberId,
-                    numberOfFailedCharges = numberFailedCharges,
-                    terminationDate = getTerminationDate(memberId),
-                    terminationReason = getTerminationReasonForMember(personFlags)
-                )
-            }
+        val terminationDate = getTerminationDate(memberId)
 
-            else -> {
-                MemberTerminationOutcome.NotTerminated(
-                    memberId = memberId,
-                    numberOfFailedCharges = numberFailedCharges,
-                    numberOfChargesLeftUntilTermination = numberOfFailedChargesRequiredForTermination(market) - numberFailedCharges
-                )
-            }
-        }
+        return if(numberFailedCharges >= numberOfFailedChargesRequiredForTermination) {
+            try {
+                val result =
+                    terminateAllContractsForMember(memberId, numberFailedCharges, terminationDate)
 
-        return when (memberTerminationOutcome) {
-            is MemberTerminationOutcome.Terminated -> {
-                activeContracts.forEach { contract ->
-                    terminateContract(
-                        contract,
+                when (result) {
+                    AllContractsTerminationState.ALL_CONTRACTS_TERMINATED -> MemberTerminationOutcome.Terminated(
                         memberId,
                         numberFailedCharges,
-                        personFlags,
-                        memberTerminationOutcome.terminationReason
+                        terminationDate,
+                        TerminationReason.MISSED_PAYMENTS
                     )
+                    AllContractsTerminationState.NOT_ALL_CONTRACTS_TERMINATED -> {
+                        logger.error("Unable to terminate ALL contracts for member $memberId")
+                        memberIsNotTerminatedOutcome(memberId, numberFailedCharges, numberOfFailedChargesRequiredForTermination)
+                    }
+                    AllContractsTerminationState.NO_CONTRACTS_TO_TERMINATE -> MemberTerminationOutcome.NoContractsToTerminate(memberId, numberFailedCharges)
                 }
-                memberTerminationOutcome
+            } catch(exception: Exception) {
+                logger.error("Error when trying to terminate all contracts for member $memberId, exception: $exception")
+                return memberIsNotTerminatedOutcome(memberId, numberFailedCharges, numberOfFailedChargesRequiredForTermination)
```

Comment:
Maybe add the helper function on the enum? Or maybe that's weird 🤷 
```suggestion
                return MemberTerminationOutcome.notTerminated(memberId, numberFailedCharges, numberOfFailedChargesRequiredForTermination)
```